data/model/oci/retriever.py

```diff
@@ -37,4 +42,14 @@ def get_blob_bytes_with_digest(self, digest):
             return None
 
         assert blob.locations is not None
-        return self.storage.get_content(blob.locations, get_layer_path(blob))
+
+        # NOTE: Some storage engines are eventually consistent, and so we add a small
+        # retry here for retrieving the blobs from storage, as they may just have been
+        # written as part of the push process.
+        for retry in range(0, RETRY_COUNT):
+            try:
+                return self.storage.get_content(blob.locations, get_layer_path(blob))
+            except IOError:
+                time.sleep(RETRY_DELAY)
+
+        return None
```

Comment:
Shouldn't it reraise whatever the last failed result of `get_content` is instead?

Reply:
No. This method should return `None` if the file is not found.